code/modules/gamemaster/actions/solar_storm.dm

```diff
@@ -21,30 +21,24 @@
 	if(isnull(base_solar_gen_rate)) base_solar_gen_rate = GLOB.solar_gen_rate
 	GLOB.solar_gen_rate = mult * base_solar_gen_rate
 
-
 /datum/gm_action/solar_storm/start()
 	..()
 	length = duration
 	command_announcement.Announce("The solar storm has reached the station. Please refain from EVA and remain inside the station until it has passed.", "Anomaly Alert")
 	adjust_solar_output(5)
 
-	while(world.time <= world.time + duration)
-		if(duration % rad_interval == 0)
-			radiate()
+	var/start_time = world.time
+
+	while(world.time <= start_time + duration)
+		sleep(rad_interval)
```

Comment:
shouldn't this while loop go in a spawn() so it doesn't sleep the entire server?